src/app/shared/services/api/search.repo.ts

```diff
@@ -1,39 +1,78 @@
+import { query } from '@angular/animations';
 import { ArchiveVO, RecordVO, FolderVO, ItemVO, TagVOData } from '@root/app/models';
 import { BaseResponse, BaseRepo } from '@shared/services/api/base';
 import { flatten } from 'lodash';
 import { Observable } from 'rxjs';
 
 export class SearchRepo extends BaseRepo {
   public archiveByEmail(email: string): Observable<SearchResponse> {
-    const data = [{
-      SearchVO: {
-        query: email
-      }
-    }];
-
-    return this.http.sendRequest<SearchResponse>('/search/archiveByEmail', data, SearchResponse);
+    const data = [
+      {
+        SearchVO: {
+          query: email,
+        },
+      },
+    ];
+
+    return this.http.sendRequest<SearchResponse>(
+      '/search/archiveByEmail',
+      data,
+      SearchResponse
+    );
   }
 
   public archiveByNameObservable(query: string): Observable<SearchResponse> {
-    const data = [{
+    const data = [
+      {
+        SearchVO: {
+          query,
+        },
+      },
+    ];
+
+    return this.http.sendRequest<SearchResponse>(
+      '/search/archive',
+      data,
+      SearchResponse
+    );
+  }
+
+  public itemsByNameObservable(
+    query: string,
+    tags: any[] = [],
+    limit?: number
+  ): Observable<SearchResponse> {
+    const data = {
       SearchVO: {
-        query
+      query,
+      tags: tags,
+      numberOfResults: limit,
       }
-    }];
+    };
 
-    return this.http.sendRequest<SearchResponse>('/search/archive', data, SearchResponse);
+    return this.http.sendV2Request<SearchResponse>(
+      '/search/folderAndRecord',
+      [data],
+      SearchResponse
+    );
   }
 
-  public itemsByNameObservable(query: string, tags: TagVOData[] = [], limit?: number): Observable<SearchResponse> {
+  public itemsByNameInPublicArchiveObservable(
+    query: string,
+    tags: any[] = [],
+    limit?: number
+  ): Observable<SearchResponse> {
     const data = {
-      SearchVO: {
-        query,
-        TagVOs: tags,
-        numberOfResults: limit
-      }
+      query,
+      tags:'',
+      archiveId: 13,
```

Comment:
I know this is in progress, but it looks like something hardcoded was left here by accident. Just mentioning it so it doesn't sneak through.

Reply:
thank you!